main.c

```diff
@@ -0,0 +1,438 @@
+            attron(COLOR_PAIR(white_green));
+        }
+        button(y, center_x, choices->values[i]->label, longest);
+        if(i == selected) {
+            attroff(COLOR_PAIR(white_green));
+        }
+    }
+}
+
+int color_byte_to_ncurses(uint8_t color_byte)
+{
+    return color_byte / 255.0 * 1000;
+}
+
+void init_color_from_bytes(short color,
+                           uint8_t byte_r, uint8_t byte_g, uint8_t byte_b)
+{
+    int nc_r = color_byte_to_ncurses(byte_r);
+    int nc_g = color_byte_to_ncurses(byte_g);
+    int nc_b = color_byte_to_ncurses(byte_b);
+    init_color(color, nc_r, nc_g, nc_b);
+}
+
+void write_output(char *fname, iso_data_t *iso_data)
+{
+    FILE *f = fopen(fname, "w");
+    if(!f) {
+        syslog(LOG_ERR, "failed to open output file [%s]", fname);
+        exit(1);
+    }
+
+    fprintf(f, "MEDIA_URL=\"%s\"\n", iso_data->url);
+    fprintf(f, "MEDIA_LABEL=\"%s\"\n", iso_data->label);
+    fprintf(f, "MEDIA_SIZE=\"%d\"\n", iso_data->size);
+    fclose(f);
+}
+
+void choice_handle_event(args_t *args, choices_t *choices, choice_event evt)
+{
+    switch(evt) {
+        case DECREASE:
+            if(choices->cur > 0) {
+                choices->cur--;
+            }
+            break;
+        case SELECT:
+            iso_data_t *cur = choices->values[choices->cur];
+            write_output(args->outfile, cur);
+            syslog(LOG_DEBUG, "selected:%s %s %d",
+                   cur->label, cur->url, cur->size);
+            break;
+        case INCREASE:
+            if(choices->cur < choices->len - 1) {
+                choices->cur++;
+            }
+            break;
+        default:
+            syslog(LOG_ERR, "invalid event id [%d]", evt);
+            exit(1);
+    }
+}
+
+#define UNUSED(X) __attribute__((unused(X)))
+
+char *find_largest_subkey(json_object *obj)
+{
+    char *ret = NULL;
+    json_object_object_foreach(obj, key, val) {
+        (void)val;
+        if(!ret || strcmp(ret, key) < 0) {
+            ret = key;
+        }
+    }
+    return ret;
+}
+
+choices_t *read_iso_choices(char *filename)
+{
+    json_object *root = json_object_from_file(filename);
+    if(!root) return NULL;
+
+    json_object *products = json_object_object_get(root, "products");
+    json_object *lunar = json_object_object_get(products,
+            "com.ubuntu.cdimage.daily:ubuntu-server:daily-live:23.04:amd64");
+    json_object *codename = json_object_object_get(lunar, "release_codename");
+    json_object *title = json_object_object_get(lunar, "release_title");
+    json_object *versions = json_object_object_get(lunar, "versions");
+    char *recent = find_largest_subkey(versions);
+    json_object *date = json_object_object_get(versions, recent);
+    json_object *items = json_object_object_get(date, "items");
+    json_object *iso = json_object_object_get(items, "iso");
+    json_object *path = json_object_object_get(iso, "path");
+    json_object *size = json_object_object_get(iso, "size");
```

Comment:
Well yeah this isn't really acceptable as you know. At the very least it needs error checking!

Reply:
I have an initial answer here that is better, and further refinements that I'll save for a follow-on.  This PR is big enough.